recipes/paho-mqtt-c/all/conanfile.py

```diff
@@ -77,6 +79,9 @@ def _configure_cmake(self):
         if self.options.ssl:
             self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
             self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+        if self.options.high_performance and tools.Version(self.version) < "1.3.2":
+            raise ConanInvalidConfiguration("{}/{} does not support -DPAHO_HIGH_PERFORMANCE flag".format(self.name, self.version))
+        self._cmake.definitions["PAHO_HIGH_PERFORMANCE"] = self.options.high_performance
```

Comment:
The correct way is to handle this option in config_options() method and completely delete this option if it's not yet available in some version.
This way it won't affect package_id for old versions which is a good thing.

Reply:
Ok, will try and look for references and will rewrite this. Thanks for the advice

Reply:
Here is a good example how to do it in the best way possible.

https://github.com/conan-io/conan-center-index/blob/7e17527f7d66e4950185694eb770c409c2b11082/recipes/sqlite3/all/conanfile.py#L77

Reply:
You create a simple property and use it in two contexts - both in `config_options` and while setting cmake variables.

Reply:
Thanks, that's a neat approach. Redone the code.